src/test/java/uk/gov/justice/hmpps/datacompliance/services/retention/RetentionServiceTest.java

```diff
@@ -109,16 +110,43 @@ void conductRetentionChecksRetentionNotRequired() {
         final var retentionChecks = service.conductRetentionChecks(OFFENDER_NUMBER);
 
         assertThat(retentionChecks).extracting(ActionableRetentionCheck::getRetentionCheck)
-                .containsExactlyInAnyOrder(
-                        new RetentionCheckPathfinder(RETENTION_NOT_REQUIRED),
-                        new RetentionCheckManual(RETENTION_NOT_REQUIRED),
-                        new RetentionCheckImageDuplicate(RETENTION_NOT_REQUIRED),
-                        new RetentionCheckDataDuplicate(PENDING));
+                .allMatch(check -> DATA_DUPLICATE.equals(check.getCheckType())
+                        ? check.getCheckStatus() == PENDING
+                        : check.getCheckStatus() == RETENTION_NOT_REQUIRED);
 
         retentionChecks.forEach(ActionableRetentionCheck::triggerPendingCheck);
         verify(dataDuplicationDetectionService).searchForDuplicates(eq(OFFENDER_NUMBER), any());
     }
 
+    @Test
+    void conductRetentionChecksWhenSqlDataDuplicateCheckDisabled() {
+
+        service = new RetentionService(
+                pathfinderApiClient,
+                manualRetentionService,
+                imageDuplicationDetectionService,
+                dataDuplicationDetectionService,
+                retentionCheckRepository,
+                referralResolutionService,
+                DataComplianceProperties.builder()
+                        .sqlDataDuplicateCheckEnabled(false)
+                        .build());
+
+        when(pathfinderApiClient.isReferredToPathfinder(OFFENDER_NUMBER)).thenReturn(false);
+        when(manualRetentionService.findManualOffenderRetentionWithReasons(OFFENDER_NUMBER)).thenReturn(Optional.empty());
+        when(imageDuplicationDetectionService.findDuplicatesFor(OFFENDER_NUMBER)).thenReturn(emptyList());
+
+        final var retentionChecks = service.conductRetentionChecks(OFFENDER_NUMBER);
+
+        assertThat(retentionChecks).extracting(ActionableRetentionCheck::getRetentionCheck)
```

Comment:
I appreciate this is not overly complicated but not a fan of logic in tests...who tests the tests.

Reply:
Yes, agreed, I didn't particularly like doing this... it stems back to JPA entities having equals and hash code based solely on the primary key, which I think we've discussed before.  In tests you can't then easily assert equality for the contents of that entity.
https://stackoverflow.com/questions/5031614/the-jpa-hashcode-equals-dilemma